### ISSUE TITLE
Custom remote rule does not show custom message if applied by addClassRules

### ISSUE THREAD

**Issue by vsorokin** [jzaefferer/jquery-validation#1390]
Let's assume we create a custom remote rule:
```javascript
$.validator.addMethod("my-remote", $.validator.methods.remote, "My message");
```
and bind it to some input using `addClassRules`:
```javascript
$.validator.addClassRules("some-class", {"my-remote": {/*config here*/}};
```
Then:
1. I enter some intentionally invalid value into the input.
2. Press Submit.
3. Input is validated, remote service is called, it returns false as expected.
4. Expected: "My message" is shown as an error message.
    Actual: Default message "Please fix this field." is shown.
5. I press Submit again - "My message" is shown as expected.
 
BTW, validation *does work* as expected if that custom rule is bound directly, not via `addClassRules`:
```javascript
        $('.form').validate({
            rules: {
                "some-input": {
                    "my-remote": {/*config here*/ }
                }
            },
            messages: {
                "some-input": {
                    "my-remote": "My message"
                }
            }
        });
```

**Comment by jzaefferer** [comment]
Thanks for the report. Due to limited resources this is unlikely to get fixed anytime soon. Considering that there's an alternative that works, it seems like a rather small issue anyway. And I suspect that `addClassRules` will go away once I, eventually, get to rewrite the whole plugin.

**Comment by vsorokin** [comment]
Hi Jörn,
Thanks for your answer.
Actually, in my app I made a workaround - my remote validation service returns error message instead of `false`, and validation plugin uses this message instead of the message given to `$.validator.addMethod(...)`.
But since you are saying you are going to remove `addClassRules` some day, could you please advise, how to solve my problem without calling it.
The problem is that we need to bind validation rules to inputs on-the-fly, we can not just enumerate them in `$('.form').validate(.....)`. How to do this without `addClassRules`?

**Comment by jzaefferer** [comment]
You're already defining an alias for the remote method, isn't that all you need? If not, you could make a custom method that delegates to the remote method, specifying whatever parameters you need.

**Comment by vsorokin** [comment]
Hi Jörn,
Let me explain our problem as briefly as I can.
We use Grails. Often, some input fields (e.g. ClientID) should be included into several different pages (and of course it should have absolutely identical validation rules, no matter where it's included into). Currently, we created template (e.g. clientId.gsp) for every such "generic" field, which looks like this:
```javascript
     <g:textField name="client-id" id="client-id"/>
     <g:javascript>
        $(document).ready(function() {
            $(".form").validate({
                rules: {
                    "client-id": "required"
                }
            });
        });
     </g:javascript>
```
Let's assume, we have several templates, such as ClientName, UserId, etc, each of them has validation rules.
Now, we insert them into the actual page:
```javascript
Please enter data:
<g:render template="clientId" />
<g:render template="clientName" />
<g:render template="userId" />
```
**And here is when we are in trouble.**
Because in result html we have **multiple validation blocks**:
```javascript
        $(document).ready(function() {
            $(".form").validate({
                rules: {
                    "client-id": "required"
                }
            });
        });
        $(document).ready(function() {
            $(".form").validate({
                rules: {
                    "client-name": "required"
                }
            });
        });
        $(document).ready(function() {
            $(".form").validate({
                rules: {
                    "user-id": "required"
                }
            });
        });
```
**Only the first `validate` works in this case**, the next ones will be ignored.

This is what I meant by "bind validation rules to inputs on-the-fly", sorry if I was unclear...

We solved this issue **by using `addClassRules` and a custom class for every field**. 
E.g. ClientID template now looks like this:
```javascript
     <g:textField name="client-id" id="client-id" class="client-id-cls"/>
     <g:javascript>
          $.validator.addClassRules("client-id-cls", {
                "required": true
            });
     </g:javascript>
```
Therefore, currently all validation rules are "encapsulated" in templates and not duplicated in actual pages.

So the question is, how to do the same properly? And, particularly, how to do it without `addClassRules` since you are going to remove it?

**Comment by jzaefferer** [comment]
Alright.

**Comment by vsorokin** [comment]
Jörn,
It seems you misunderstood me.
I meant I managed to dispense with `addClassRules` by using `rules add`, so this particular issue is not a problem *for me*.
But this is still a bug, even though it may be insignificant.
So I think it is too early to close it.
